modules/core/src/passes/layers-pass.ts

```diff
@@ -2,17 +2,56 @@ import GL from '@luma.gl/constants';
 import Pass from './pass';
 import {clear, setParameters, withParameters, cssToDeviceRatio} from '@luma.gl/core';
 
+import type {Framebuffer} from '@luma.gl/core';
+import type Viewport from '../viewports/viewport';
+import type View from '../views/view';
+import type Layer from '../lib/layer';
+import type Effect from '../lib/effect';
+
+export type LayersPassRenderOptions = {
+  target?: Framebuffer;
+  pass: string;
+  layers: Layer[];
+  viewports: Viewport[];
+  onViewportActive: (viewport: Viewport) => void;
+  views?: Record<string, View>;
+  effects?: Effect[];
+  clearCanvas?: boolean;
+  layerFilter?: (context: FilterContext) => boolean;
+  moduleParameters?: any;
+};
+
+type DrawLayerParameters = {
+  shouldDrawLayer: boolean;
+  layerRenderIndex: number;
+  moduleParameters?: any;
+  layerParameters?: any;
+};
+
+type FilterContext = {
+  layer: Layer;
+  viewport: Viewport;
+  isPicking: boolean;
+  renderPass: string;
+};
+
+export type RenderStats = {
+  totalCount: number;
+  visibleCount: number;
+  compositeCount: number;
+  pickableCount: number;
+};
+
 export default class LayersPass extends Pass {
-  render(props) {
+  render(props: LayersPassRenderOptions): any {
     const gl = this.gl;
 
     setParameters(gl, {framebuffer: props.target});
     return this._drawLayers(props);
   }
 
-  // PRIVATE
   // Draw a list of layers in a list of viewports
-  _drawLayers(props) {
+  private _drawLayers(props: LayersPassRenderOptions) {
```

Comment:
Nit: separate `options` and `props`? `props` in constructor and options for named arguments to methods?
```suggestion
  private _drawLayers(options: LayersPassRenderOptions) {
```